### PR TITLE
unify mobilenetv2 namespace, fix output/activation logic

### PR DESCRIPTION
**Pull request by brettkoonce**

### CODE REVIEW

File: Models/ImageClassification/MobileNetV2.swift [hunk 1]
```diff
@@ -120,38 +120,47 @@ public struct MobileNetV2: Layer {
...
-            initialInvertedBottleneck)
-        let backbone = convolved.sequenced(through: residualBlockStack1, residualBlockStack2,
+        let convolved = relu6(input.sequenced(through: inputConv, inputConvBatchNorm))
+        let initialConv = initialInvertedBottleneck(convolved)
+
+        let backbone = initialConv.sequenced(through: residualBlockStack1, residualBlockStack2,
             residualBlockStack3, residualBlockStack4, residualBlockStack5)
-        return backbone.sequenced(through: invertedBottleneckBlock16, finalConv, avgPool, output)
+        let output = relu6(outputConvBatchNorm(outputConv(invertedBottleneckBlock16(backbone))))
+        return output.sequenced(through: avgPool, dropoutProb, outputClassifer)
```
Comment (texasmichelle) [inline Models/ImageClassification/MobileNetV2.swift]: Why add these two layers: `dropoutProb` and `outputClassifier`?

File: Models/ImageClassification/MobileNetV2.swift [hunk 2]
```diff
@@ -84,9 +84,9 @@ public struct InvertedResidualBlock: Layer {
         let pw = relu6(batchNormConv1(conv1(input)))
         var dw: Tensor<Float>
         if self.strides == (1, 1) {
-            dw = batchNormDConv(dConv(pw))
+            dw = relu6(batchNormDConv(dConv(pw)))
         } else {
-            dw = zeroPad(batchNormDConv(dConv(pw)))
+            dw = relu6(zeroPad(batchNormDConv(dConv(pw))))
```
[the remark this answers is not in the document]
Comment (brettkoonce) [reply Models/ImageClassification/MobileNetV2.swift]: i've been staring at this zeropad logic for a few days (i do not trust it either!), is there a specific implementation you are going off of?  happy to make the change, just haven't found a good ground source of truth...

File: Models/ImageClassification/MobileNetV2.swift [hunk 3]
```diff
@@ -84,9 +84,9 @@ public struct InvertedResidualBlock: Layer {
         let pw = relu6(batchNormConv1(conv1(input)))
```
[the remark this answers is not in the document]
Comment (brettkoonce) [reply Models/ImageClassification/MobileNetV2.swift]: sure, one sec

File: Models/ImageClassification/MobileNetV2.swift (hunk 1 above)
Comment (brettkoonce) [reply Models/ImageClassification/MobileNetV2.swift]: i've been copying the pytorch implementation here, that's how they do it: https://github.com/pytorch/vision/blob/3c254fb7af5f8af252c24e89949c54a3461ff0be/torchvision/models/mobilenet.py#L132

File: Models/ImageClassification/MobileNetV2.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (texasmichelle) [reply Models/ImageClassification/MobileNetV2.swift]: I'm going off the Keras implementation. In [this line](https://github.com/keras-team/keras-applications/blob/master/keras_applications/mobilenet_v2.py#L452), it comes before `dConv`, `batchNormDConv`, and `relu6`.

File: Models/ImageClassification/MobileNetV2.swift (hunk 1 above)
Comment (texasmichelle) [reply Models/ImageClassification/MobileNetV2.swift]: That looks like a separate model rather than additional layers (classifier vs features). Let's stick with keras as a reference implementation since that's what I used for mobilenetv1.

**Comment by texasmichelle** [comment]
Two other cleanup items that would be worth making:
* Add `widthMultiplier` (`alpha` in Keras)
* Change the filename from MobileNet.swift -> MobileNetV1.swift

File: Models/ImageClassification/MobileNetV2.swift (hunk 1 above)
Comment (texasmichelle) [reply Models/ImageClassification/MobileNetV2.swift]: Oh, I see they're combined at the end... hm. I still think we should use Keras to compare.

File: Models/ImageClassification/MobileNetV2.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (brettkoonce) [reply Models/ImageClassification/MobileNetV2.swift]: thanks for the link, updating!

File: Models/ImageClassification/MobileNetV2.swift [hunk 4]
```diff
@@ -120,38 +120,47 @@ public struct MobileNetV2: Layer {
...
+        dropoutProb = Dropout<Float>(probability: dropout)
+        outputClassifer = Dense(inputSize: 1280, outputSize: classCount)
     }
 
     @differentiable
     public func callAsFunction(_ input: Tensor<Float>) -> Tensor<Float> {
-        let convolved = input.sequenced(through: inputConv, inputConvBatchNorm,
-            initialInvertedBottleneck)
-        let backbone = convolved.sequenced(through: residualBlockStack1, residualBlockStack2,
+        let convolved = relu6(input.sequenced(through: inputConv, inputConvBatchNorm))
```
[the remark this answers is not in the document]
Comment (brettkoonce) [reply Models/ImageClassification/MobileNetV2.swift]: kk, https://github.com/keras-team/keras-applications/blob/b34c10628a0ab436542e9160f98de72b49084bbe/keras_applications/mobilenet_v2.py#L310

File: Models/ImageClassification/MobileNetV2.swift (hunk 1 above)
Comment (brettkoonce) [reply Models/ImageClassification/MobileNetV2.swift]: kk, matched!

**Comment by brettkoonce** [comment]
@texasmichelle think that gets it, i'll port these changes back to #394 and #409 next!

**Comment by brettkoonce** [comment]
@texasmichelle above has regressed to 50% on imagenette, was 70% or so before...

**Comment by brettkoonce** [comment]
softmax is the culprit it seems

**Comment by brettkoonce** [comment]
@texasmichelle false alarm, it's just more sensitive to learning rate now is all, converges fine w/ lr = 0.002

**Comment by brettkoonce** [comment]
working with lower learning rates here!

**Comment by texasmichelle** [comment]
This is great progress!

I found [this example](https://www.tensorflow.org/tutorials/images/transfer_learning), which comes with a highly useful Colab notebook. It explains what the extra layers are for, among other things. I recommend running the notebook as a python source of truth. In particular, it would be good to verify architectures. We don't have a `model.summary()` equivalent yet, but injecting print statements with type and shape will get you most of the way there and show whether each layer is in the right order.

File: Models/ImageClassification/MobileNetV2.swift [hunk 5]
```diff
@@ -116,42 +148,73 @@ public struct InvertedResidualBlockStack: Layer {
...
-    public var invertedBottleneckBlock16 = InvertedResidualBlock(filters: (160, 320))
-    public var finalConv: Conv2D<Float>
+    public var outputConv: Conv2D<Float>
+    public var outputConvBatchNorm: BatchNorm<Float>
     public var avgPool = GlobalAvgPool2D<Float>()
-    public var output: Dense<Float>
+    public var outputClassifer: Dense<Float>
 
-    public init(classCount: Int = 1000) {
+    public init(classCount: Int = 1000, width: Float = 1.0) {
```
Comment (texasmichelle) [inline Models/ImageClassification/MobileNetV2.swift]: nit: s/width/widthMultiplier/ for consistency with MobileNetV1

File: Models/ImageClassification/MobileNetV2.swift [hunk 6]
```diff
@@ -19,33 +19,53 @@ import TensorFlow
 // Mark Sandler, Andrew Howard, Menglong Zhu, Andrey Zhmoginov, Liang-Chieh Chen
 // https://arxiv.org/abs/1801.04381
 
+fileprivate func makeDivisible(filter: Int, width: Float = 1.0, divisor: Float = 8.0) -> Int {
```
Comment (texasmichelle) [inline Models/ImageClassification/MobileNetV2.swift]: This file could use a run through swift-format

File: Models/ImageClassification/MobileNetV2.swift (hunk 5 above)
Comment (brettkoonce) [reply Models/ImageClassification/MobileNetV2.swift]: done!

File: Models/ImageClassification/MobileNetV2.swift (hunk 6 above)
Comment (brettkoonce) [reply Models/ImageClassification/MobileNetV2.swift]: done!

File: Tests/ImageClassificationTests/Inference.swift [hunk 7]
```diff
@@ -81,6 +81,16 @@ final class ImageClassificationInferenceTests: XCTestCase {
         let mobileNet = MobileNetV2(classCount: 1000)
         let mobileNetResult = mobileNet(inputImageNet)
         XCTAssertEqual(mobileNetResult.shape, [1, 1000])
+
+        // Width multiplier
+        let mobileNet2WMSmall = MobileNetV2(classCount: 10, width: 0.5)
```
Comment (texasmichelle) [inline Tests/ImageClassificationTests/Inference.swift]: s/width/widthMultiplier/
In both tests

**Comment by texasmichelle** [comment]
Thank you for making these changes!

How are you running this? Can you commit your example file?

File: Tests/ImageClassificationTests/Inference.swift (hunk 7 above)
Comment (brettkoonce) [reply Tests/ImageClassificationTests/Inference.swift]: doh, fixed!

File: Examples/MobileNet-Imagenette/main.swift [hunk 8]
```diff
@@ -19,17 +19,17 @@ import TensorFlow
 let batchers = ImagenetteBatchers(
     inputSize: .resized320, 
     outputSize: 224, 
-    batchSize: 64,
+    batchSize: 16,
     numWorkers: 8
 )
 
-var model = MobileNetV1(classCount: 10)
+var model = MobileNetV2(classCount: 10)
```
Comment (brettkoonce) [inline Examples/MobileNet-Imagenette/main.swift]: @texasmichelle here you go!

**Comment by brettkoonce** [comment]
gonna need a rebase soon 😅

**Comment by brettkoonce** [comment]
eg will back out harness then

**Comment by brettkoonce** [comment]
@texasmichelle here's a clean rebase!
re example: if nobody's in love with the imagewoof demo, i could swap this model in there!

**Comment by texasmichelle** [comment]
Imagewoof sounds like a good fit for this. Is there value in having both VGG and MobileNetV2 with that dataset? Examples are small enough that there's no harm in demonstrating both.

**Comment by brettkoonce** [comment]
imagewoof is being a bit of pain. just duped the existing v1 demo and put in the hyperparameters from before.  renamed the mn v1 stuff to match, ran everything through the linter.

File: Package.swift [hunk 9]
```diff
@@ -18,11 +18,12 @@ let package = Package(
         .library(name: "TextModels", targets: ["TextModels"]),
         .executable(name: "Benchmarks", targets: ["Benchmarks"]),
         .executable(name: "VGG-Imagewoof", targets: ["VGG-Imagewoof"]),
+        .executable(name: "MobileNetV2-Imagenette", targets: ["MobileNetV2-Imagenette"]),
```
Comment (texasmichelle) [inline Package.swift]: nit: organize this line underneath MobileNetV1

File: Package.swift [hunk 10]
```diff
@@ -58,6 +60,9 @@ let package = Package(
         .target(
             name: "VGG-Imagewoof", dependencies: ["ImageClassificationModels", "Datasets"],
             path: "Examples/VGG-Imagewoof"),
+        .target(
```
Comment (texasmichelle) [inline Package.swift]: same organizational nit

**Comment by texasmichelle** [comment]
This looks great!! Thank you for these updates 🎉

File: Package.swift (hunk 9 above)
Comment (brettkoonce) [reply Package.swift]: see updated!

File: Package.swift (hunk 10 above)
Comment (brettkoonce) [reply Package.swift]: see updated!

**Comment by brettkoonce** [comment]
@texasmichelle plz to try again k thx
